src/stages/main/patchers/AssignOpPatcher.js

```diff
@@ -17,14 +19,123 @@ export default class AssignOpPatcher extends NodePatcher {
   }
 
   patchAsExpression() {
-    this.assignee.patch();
-    this.expression.patch();
+    if (this.isExpansionAssignment()) {
+      this.patchExpansionAssignment();
+    } else {
+      this.assignee.patch();
+      this.expression.patch();
+    }
+  }
+
+  statementNeedsParens(): boolean {
+    if (this.isExpansionAssignment()) {
+      return this.expansionAssignmentNeedsParens();
+    } else {
+      // The assignment needs parentheses when the LHS needs parens.
+      return this.assignee.statementShouldAddParens();
+    }
+  }
+
+  expansionAssignmentNeedsParens(): boolean {
+    if (!this.expression.isRepeatable()) {
+      // The left side will be an "array" variable.
+      return false;
+    }
+    let expansionIndex = this.getExpansionIndex();
+    if (expansionIndex === this.assignee.members.length - 1) {
+      // Simple case where we leave the array assignment mostly intact.
+      return this.assignee.statementShouldAddParens();
+    } else if (expansionIndex === 0) {
+      // The first non-expansion assignee will end up on the left side.
+      return this.assignee.members[1].statementShouldAddParens();
+    } else {
+      return this.assignee.members[0].statementShouldAddParens();
+    }
+  }
+
+  isExpansionAssignment(): boolean {
+    return this.getExpansionIndex() !== -1;
   }
 
   /**
-   * The assignment needs parentheses when the LHS needs parens.
+   * If there is an expansion assignment, return the index of the expansion node.
+   * Otherwise, return -1.
    */
-  statementNeedsParens(): boolean {
-    return this.assignee.statementShouldAddParens();
+  getExpansionIndex(): number {
+    if (!(this.assignee instanceof ArrayInitialiserPatcher)) {
+      return -1;
+    }
+    for (let i = 0; i < this.assignee.members.length; i++) {
+      if (this.assignee.members[i] instanceof ExpansionPatcher) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  patchExpansionAssignment() {
+    let expansionIndex = this.getExpansionIndex();
+    let assignees = this.assignee.members;
+    let expansionNode = assignees[expansionIndex];
+
+    assignees.forEach((assignee, i) => {
+      // Patch everything but the expansion node, since expansion nodes expect
+      // to not be patched.
+      if (i !== expansionIndex) {
+        assignee.patch();
+      }
+    });
+    this.expression.patch();
+    let expressionCode = this.slice(this.expression.contentStart, this.expression.contentEnd);
```

Comment:
This can be done using `this.expression.getOriginalCode()`. It's not exactly the same in all cases since `this.slice()` uses the patched string. In this case `this.expression` shouldn't have been patched yet so it should be the same.